### PR TITLE
Feat: Add React support and improve exported config

### PR DESCRIPTION
**Pull request by ncomerci**

### CODE REVIEW

File: yarn-error.log
Comment (facuspagnuolo) [inline yarn-error.log:1]: Let's add this to ignore file

File: README.md [hunk 1]
```diff
@@ -44,6 +44,78 @@ $ cd eslint-mimic-config
...
+
+- **Node.js** version `>=18.0.0`
+- **ESLint** version `>=8.21.0`
+
+### Installation
+
+Install the `eslint-config-mimic` package as a development dependency:
+
+```bash
+# Using npm
```
Comment (facuspagnuolo) [inline README.md]: We can skip npm as we don't encourage it from Mimic

File: package.json [hunk 2]
```diff
@@ -1,25 +1,28 @@
 {
   "name": "eslint-config-mimic",
-  "version": "0.0.2",
+  "version": "0.0.3",
```
Comment (facuspagnuolo) [inline package.json]: Let's do the bump with the new version tag on a separate commit
```suggestion
  "version": "0.0.2",
```

File: index.mjs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      "import/resolver": {
+        typescript: {
+          alwaysTryTypes: true,
+          project: "./tsconfig.json",
+        },
+      },
+    },
+  },
+  eslintConfigPrettier,
+];
```
Comment (facuspagnuolo) [inline index.mjs]:
```suggestion
]

```

File: README.md [hunk 4]
```diff
@@ -9,29 +9,29 @@
     <img src="https://badge.fury.io/js/eslint-config-mimic.svg" alt="NPM">
   </a>
   <a href="https://discord.mimic.fi">
-    <img src="https://img.shields.io/discourse/status?server=https%3A%2F%2Fmeta.discourse.org" alt="Discord">
+    <img src="https://img.shields.io/discord/747115528198332456" alt="Discord">
```
Comment (facuspagnuolo) [inline README.md]: This one seems invalid, see:

https://img.shields.io/discord/747115528198332456

**Review by facuspagnuolo — COMMENTED** [review]
Curious, how would this work with [solhint-mimic-config](https://github.com/mimic-fi/solhint-mimic-config)? 

For example, in v3-core we use it (see https://github.com/mimic-fi/v3-core/blob/master/packages/authorizer/.prettierrc.js)

**Comment by ncomerci** [comment]
> Curious, how would this work with [solhint-mimic-config](https://github.com/mimic-fi/solhint-mimic-config)?
> 
> For example, in v3-core we use it (see https://github.com/mimic-fi/v3-core/blob/master/packages/authorizer/.prettierrc.js)

There should be no conflict as these rules would only apply to javascript or typescript files!

```
// index.mjs
files: [“**/*.{js,jsx,ts,tsx}”]
```

**Review by facuspagnuolo — APPROVED** [review]

**Pull request merged**